OBSDOCS-1730: Port Configuring the logging collector to 5.8 and 6.y docs (#89176)
openshift/openshift-docs: modules/log6x-log-collector-syslog-server.adoc

// Module included in the following assemblies:
//
// * observability/logging/log_collection_forwarding/cluster-logging-collector.adoc

:_mod-docs-content-type: PROCEDURE
[id="log-collector-syslog-server_{context}"]
= Configuring the collector to listen for connections as a syslog server

You can configure your log collector to collect journal format infrastructure logs by specifying `syslog` as a receiver input in the `ClusterLogForwarder` custom resource (CR).

:feature-name: Syslog receiver input
include::snippets/logging-http-sys-input-support.adoc[]

.Prerequisites

* You have administrator permissions.
* You have installed the {oc-first}.
* You have installed the {clo}.
* You have created a `ClusterLogForwarder` CR.

.Procedure

> . Grant the `collect-infrastructure-logs` cluster roles to the service account:

Reviewer: ```suggestion
. Grant the `collect-infrastructure-logs` cluster role to the service account by running the following command:
```

- Should it be "role"? if not ignore this comment.
- Add "by running the...command:"

Author: should be role, good catch. Updated

+
.Example binding command
[source,terminal]
----
$ oc adm policy add-cluster-role-to-user collect-infrastructure-logs -z logcollector
----

. Modify the `ClusterLogForwarder` CR to add configuration for the `syslog` receiver input:
+
.Example `ClusterLogForwarder` CR
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
# ...
spec:
  serviceAccount:
    name: <service_account_name>
  inputs:
    - name: syslog-receiver # <1>
      type: receiver
      receiver:
        type: syslog # <2>
        port: 10514 # <3>
  outputs:
  - name: default-lokistack
    lokiStack:
      authentication:
        token:
          from: serviceAccount
      target:
        name: logging-loki
        namespace: openshift-logging
    tls:
      ca:
        key: service-ca.crt
        configMapName: openshift-service-ca.crt
    type: lokiStack
# ...      
  pipelines: # <4>
    - name: syslog-pipeline
      inputRefs:
        - syslog-receiver
      outputRefs:
        - <output_name>
# ...
----
<1> Specify a name for your input receiver.
<2> Specify the input receiver type as `syslog`.

> <3> Optional: Specify the port that the input receiver listens on. This must be a value between `1024` and `65535`. The default value is `10514` if this is not specified.

Reviewer: ```suggestion
<3> Optional: Specify the port that the input receiver listens on. This must be a value between `1024` and `65535`. The default value is `10514`.
```

- Same comment as earlier. "if this is not specified" is intended.

<4> Configure a pipeline for your input receiver.

. Apply the changes to the `ClusterLogForwarder` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

.Verification

> * Verify that the collector is listening on the service that has a name in the format `<ClusterLogForwarder_CR_name>-<input_name>`. In this example, the service name is `collector-syslog-receiver`:

Reviewer: ```suggestion
* Verify that the collector is listening on the service that has a name in the `<clusterlogforwarder_resource_name>-<input_name>` format by running the following command:
```

Same comments as earlier

+
[source,terminal]
----
$ oc get svc
----
+
.Example output
+
[source,terminal]
----
NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)            AGE
collector                   ClusterIP   172.30.85.239    <none>        24231/TCP          33m
collector-syslog-receiver   ClusterIP   172.30.216.142   <none>        10514/TCP          2m20s

> ----

Reviewer: ```suggestion
----
+
In this example output, the service name is `collector-syslog-receiver`.
```

Same comment as earlier.

// Module included in the following assemblies:
//
// * observability/logging/log_collection_forwarding/cluster-logging-collector.adoc

:_mod-docs-content-type: PROCEDURE
[id="log-collector-syslog-server_{context}"]
= Configuring the collector to listen for connections as a syslog server

You can configure your log collector to collect journal format infrastructure logs by specifying `syslog` as a receiver input in the `ClusterLogForwarder` custom resource (CR).

:feature-name: Syslog receiver input
include::snippets/logging-http-sys-input-support.adoc[]

.Prerequisites

* You have administrator permissions.
* You have installed the {oc-first}.
* You have installed the {clo}.
* You have created a `ClusterLogForwarder` CR.

.Procedure

. Modify the `ClusterLogForwarder` CR to add configuration for the `syslog` receiver input:
+
.Example `ClusterLogForwarder` CR
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
# ...
spec:

>   inputs:

Reviewer 2: Please specify the seviceacount. 
  serviceAccount:
    name: logcollector

please indicate we must grant collect-infrastructure-logs  to this account.

"oc adm policy add-cluster-role-to-user collect-infrastructure-logs -z logcollector"

Reviewer 3: I think we have this info available here https://docs.openshift.com/container-platform/4.16/observability/logging/logging-6.0/log6x-clf.html#creating-service-accounts itself.

Reviewer 2: Yes. that is the pre-request  for inputs.syslog. 
For inputs.http,  the serviceaccount needn't any special roles.
For inputs.syslog, the serviceaccount must be granted collect-infrastructure-logs.

Author: added

    - name: syslog-receiver # <1>
      type: receiver
      receiver:
        type: syslog # <2>
        port: 10514 # <3>

>         syslog:

Reviewer 3: Remove line 40-41 as those are not a part of syslog config.

          format: kubeAPIAudit   
  pipelines: # <4>
    - name: syslog-pipeline
      inputRefs:
        - syslog-receiver
# ...
----
<1> Specify a name for your input receiver.
<2> Specify the input receiver type as `syslog`.
<3> Optional: Specify the port that the input receiver listens on. This must be a value between `1024` and `65535`. The default value is `10514` if this is not specified.
<4> Configure a pipeline for your input receiver.

. Apply the changes to the `ClusterLogForwarder` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

> 

Reviewer 2: Can you add the expected result?  Once CLF is created, the collector listen on service <clf-name>-<inputs.[x].name>
```
$ oc get svc
NAME                        TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)            AGE
collector                   ClusterIP   172.30.85.239    <none>        24231/TCP          33m
collector-syslog-receiver   ClusterIP   172.30.216.142   <none>        10514/TCP          2m20s
```

----

. Modify the `ClusterLogForwarder` CR to add configuration for the `syslog` receiver input:
+
.Example `ClusterLogForwarder` CR
[source,yaml]
----
apiVersion: observability.openshift.io/v1
kind: ClusterLogForwarder
metadata:
# ...
spec:
  serviceAccount:
    name: <service_account_name>
  inputs:
    - name: syslog-receiver # <1>
      type: receiver
      receiver:
        type: syslog # <2>
        port: 10514 # <3>
  outputs:
  - name: <output_name>
    type: <output-type>
    <output-type>:
      url: <log_output_url>
# ...      
  pipelines: # <4>
    - name: syslog-pipeline
      inputRefs:
        - syslog-receiver
      outputRefs:
        - <output_name>
# ...
----
<1> Specify a name for your input receiver.
<2> Specify the input receiver type as `syslog`.
<3> Optional: Specify the port that the input receiver listens on. This must be a value between `1024` and `65535`. The default value is `10514` if this is not specified.
<4> Configure a pipeline for your input receiver.

. Apply the changes to the `ClusterLogForwarder` CR by running the following command:
+
[source,terminal]
----
$ oc apply -f <filename>.yaml
----

.Verification

> * Verify that the collector is listening on the service that has a name in the format `e <ClusterLogForwarder_CR_name>-<input_name>`. In this example, the service name is `collector-syslog-receiver`:

Author: sorry, fixed

.Example `ClusterLogForwarder` CR if you are using a multi log forwarder deployment
[source,yaml]
----

> apiVersion: logging.openshift.io/

Author: updated the example
